reactor-netty5-http/src/main/java/reactor/netty5/http/server/HttpServerFormDecoderProvider.java

```diff
@@ -296,4 +335,120 @@ HttpServerFormDecoderProvider build() {
 			return new HttpServerFormDecoderProvider(this);
 		}
 	}
+
+	static final class ReactorNettyHttpPostMultipartRequestDecoder extends HttpPostMultipartRequestDecoder
+			implements ReactorNettyHttpPostRequestDecoder {
+
+		/**
+		 * Current {@link HttpData} from the body (only the completed {@link HttpData}).
+		 */
+		final List<HttpData> currentCompletedHttpData = new ArrayList<>();
+
+		ReactorNettyHttpPostMultipartRequestDecoder(HttpDataFactory factory, HttpRequest request) {
+			super(factory, request);
+		}
+
+		@Override
+		protected void addHttpData(InterfaceHttpData data) {
+			if (data instanceof HttpData httpData) {
+				currentCompletedHttpData.add(httpData);
+			}
+		}
+
+		@Override
+		public void cleanCurrentHttpData(boolean onlyCompleted) {
+			for (HttpData data : currentCompletedHttpData) {
+				removeHttpDataFromClean(data);
+				data.close();
+			}
+			currentCompletedHttpData.clear();
+
+			if (!onlyCompleted) {
+				InterfaceHttpData partial = currentPartialHttpData();
+				if (partial instanceof HttpData partialData) {
+					partialData.delete();
+				}
+			}
+		}
+
+		@Override
+		public List<HttpData> currentHttpData(boolean onlyCompleted) {
+			if (!onlyCompleted) {
+				InterfaceHttpData partial = currentPartialHttpData();
+				if (partial instanceof HttpData partialData) {
+					partialData.usingContent(partialContent ->
+							currentCompletedHttpData.add(partialData.replace(partialContent.split())));
+				}
+			}
+
+			return currentCompletedHttpData;
+		}
+
+		@Override
+		public void destroy() {
+			super.destroy();
+			InterfaceHttpData partial = currentPartialHttpData();
+			if (partial != null && partial.isAccessible()) {
```

Comment:
Why is it necessary to check for accessibility? Who owns the data?

Reply:
it's not necessary to check accessibility, because the ReactorNettyHttpPostMultipartRequestDecoder is owning the buffer, I have removed the useless check in commit [618035a](https://github.com/reactor/reactor-netty/pull/2327/commits/618035a9ee6056d639e2c277f48dde7734c647f4)